config/testdata/conf.good.yml

```diff
@@ -73,7 +73,7 @@ scrape_configs:
     target_label:  cde
   - replacement:   static
     target_label:  abc
-  - regex:
+  - regex: ""
```

Comment:
Why did this change? It's important that the previous version also is parsed, as we've had panics from things like that in the past.

Reply:
It seems like the default value of regex when nothing is mentioned is `(.*)`. That's why I had to apply an empty string.

Reply:
That sounds like a regression then, I'd expect this to be parsed as the empty string as that's what's in the YAML.

Reply:
It is dependent on regex lib most probably, since I have seen yaml v3 parsing empty as someone does expect.

I see one way as a workaround. We can make `Regex` a string and a new private member `reg` that contains the actual reg expr which will be used. This gives us the conditional control on regex and an help solve the regression.

Reply:
Well, the thing is that if we leave `regex` field empty, then after parsing, it gets a value of `null`. But, we use `DefaultConfig` where it is initialized to `(.*)` which leads to failing.

Reply:
That's a regression that needs fixing then.

Don't silently change unittest behaviour in PRs like this, this would have been a breaking change.

Reply:
Should I just keep yaml package of the particular func to v2? Since, rest all places the shift to v3 seems a success.

Reply:
We should be switching everything to v3, having bits here and there will only cause confusion long term.

Reply:
Then I will be happy if I can get some suggestions to solve the issue related to this particular case only(i.e., `Load()` in `config.go`). Maybe, `help wanted` tag if you don't have time for this.

Reply:
I opened https://github.com/go-yaml/yaml/pull/689 on the go-yaml side for this, I updated a branch of this PR locally and aside from this issue I don't see any major blockers now.